### PR TITLE
Ensure backward compatibility of ORTModel

### PR DESCRIPTION
**Pull request by fxmarty**
Following https://github.com/huggingface/optimum/pull/872, we make sure we have no breaking change for the existing exported models.

Thank you @echarlaix for pointing it out!

### CODE REVIEW

File: optimum/onnxruntime/base.py [hunk 1]
```diff
@@ -569,10 +577,21 @@ def filter_out_output(output_name):
                 out_past_key_values[i : i + self.num_pkv] for i in range(0, len(out_past_key_values), self.num_pkv)
             )
         else:
-            # grab the cross attention key/values from the inputs
-            out_past_key_values = tuple(
-                out_past_key_values[i : i + self.num_pkv] + past_key_values[2 * i + 2 : 2 * i + 2 + self.num_pkv]
-                for i in range(0, len(out_past_key_values), self.num_pkv)
-            )
+            if self.legacy_outputs is True:
```
Comment (michaelbenayoun) [inline optimum/onnxruntime/base.py:580]: :)

File: optimum/onnxruntime/base.py [hunk 2]
```diff
@@ -569,10 +577,21 @@ def filter_out_output(output_name):
...
+                    "For the decoder with past, using ONNX models outputting cross attention past key values"
+                    " is deprecated and the support will be removed in optimum 2.0. We recommend exporting again the model"
+                    " with optimum>=1.7.3."
+                )
+                warn_once(logger, msg=msg)
+                out_past_key_values = tuple(
+                    out_past_key_values[i : i + self.num_pkv] for i in range(0, len(out_past_key_values), self.num_pkv)
+                )
+            else:
+                # grab the cross attention key/values from the inputs
```
Comment (michaelbenayoun) [inline optimum/onnxruntime/base.py:591]:
```suggestion
                # Grab the cross attention key/values from the inputs
```

File: optimum/utils/logging.py [hunk 3]
```diff
@@ -262,3 +267,8 @@ def reset_format() -> None:
 
     for handler in handlers:
         handler.setFormatter(None)
+
+
+@lru_cache(None)
```
Comment (michaelbenayoun) [inline optimum/utils/logging.py:272]:
```suggestion
@lru_cache
```
is enough no?

**Review by michaelbenayoun — APPROVED** [review]
LGTM

File: optimum/utils/logging.py (hunk 3 above)
Comment (fxmarty) [reply optimum/utils/logging.py:272]: No, the default is 128: https://docs.python.org/3/library/functools.html#functools.lru_cache

**Pull request merged**